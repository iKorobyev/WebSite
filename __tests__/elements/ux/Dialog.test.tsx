import React from 'react'
import { cleanup, render, waitFor, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Dialog } from '~ux'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Dialog', () => {
  it('renders correctly', async () => {
    const props = {
      show: true,
      children: 'Test',
      onHide: jest.fn()
    }

    const { container } = render(<Dialog {...props} />)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
  })
})
