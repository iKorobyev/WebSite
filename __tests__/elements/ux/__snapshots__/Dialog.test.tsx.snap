// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog renders correctly 1`] = `<div />`;
